esp-hal/src/rtc_cntl/mod.rs

```diff
@@ -217,8 +223,18 @@ impl<'d> Rtc<'d> {
         RtcClock::estimate_xtal_frequency()
     }
 
-    /// Read the current value of the rtc time registers.
+    #[deprecated = "This function does not take into account the boot time registers, and therefore will not react to using `set_time_us` and `set_time_ms`."]
```

Comment:
I thought it should be deprecated since it isn't consistent with get_time_us and get_time_ms. What do you think about deprecating get_time_raw and then exposing get_rtc_time_raw (and possibly the other get_rtc_time functions)? At that point we could also expose the boot time functions.

Doing what you suggested also seems fine but I thought it would be best to alert users of it not respecting boot time to avoid confusion.

Reply:
No one every would want to do that, Why? make it at least a bit platform agnostic...

Reply:
I mean, you never know. One reason I can think of is to work around the overflow/wrap around issue by using more store registers. This PR only uses two of them for a 64 bit time, but someone may want to use four of them for a 128 bit time. Or maybe reduce the granularity from micro seconds to milliseconds. Really, those STORE registers can be used for anything and IDF just happened to store the boot time in there.

Reply:
I do agree that it would be good to expose the rtc time in some way in case the user doesn't want to use esp-hal's boot time implementation. If you think leaving get_time_raw undeprecated won't cause any confusion as to why it doesn't react to setting the time, that's fine with me. Although, it the goal is exposing the raw rtc time, it would likely be more beneficial to expose get_rtc_time_ms and get_rtc_time_us so the user doesn't have to do those conversions themselves.

Reply:
> Although, it the goal is exposing the raw rtc time, it would likely be more beneficial to expose get_rtc_time_ms and get_rtc_time_us so the user doesn't have to do those conversions themselves.

SGTM

Reply:
Should get_time_raw be removed and get_rtc_time_raw be exposed?

Reply:
Looks like they do the same thing. Either works I guess. Given the struct is already called Rtc, adding `rtc` in the method name seem redundant to me.

Reply:
I'm going to remove get_time_raw and expose get_rtc_time_raw. While having rtc in the method name is a bit redundant, I think it's necessary to highlight the difference between it and the get_time functions (the former not respecting boot time)